packages/client.api/src/actions/Actions.ts

```diff
@@ -43,53 +33,29 @@ export type ApplyActionOptions =
 
 export type ApplyBatchActionOptions = { $returnEdits?: boolean };
 
-type BaseType<APD extends ActionParameterDefinition<any, any>> =
-  APD["type"] extends ObjectActionDataType<any, infer TTargetType> ?
-      | OsdkBase<TTargetType>
-      | OsdkObjectPrimaryKeyType<TTargetType>
-    : APD["type"] extends ObjectSetActionDataType<any, infer TTargetType>
-      ? BaseObjectSet<TTargetType>
-    : APD["type"] extends keyof DataValueClientToWire
-      ? DataValueClientToWire[APD["type"]]
-    : never;
-
-type MaybeArrayType<APD extends ActionParameterDefinition<any, any>> =
-  APD["multiplicity"] extends true ? Array<BaseType<APD>>
-    : BaseType<APD>;
-
-type NotOptionalParams<X extends ActionParametersDefinition> = {
-  [P in keyof X]: MaybeArrayType<X[P]>;
-};
-
-export type OsdkActionParameters<
-  X extends ActionParametersDefinition,
-> = NullableProps<X> extends never ? NotOptionalParams<X>
-  : PartialBy<NotOptionalParams<X>, NullableProps<X>>;
-
-export type ActionSignatureFromDef<T extends ActionDefinition<any, any, any>> =
-  {
-    applyAction: NonNullable<T["__OsdkActionType"]> extends never
-      ? ActionSignature<T["parameters"]>
-      : NonNullable<T["__OsdkActionType"]>;
-  };
-
-type ActionParametersDefinition = Record<
-  any,
-  ActionParameterDefinition<any, any>
->;
-
-export type ActionSignature<
-  X extends Record<any, ActionParameterDefinition<any, any>>,
-> = <
-  A extends NOOP<OsdkActionParameters<X>> | NOOP<OsdkActionParameters<X>>[],
-  OP extends A extends NOOP<OsdkActionParameters<X>>[] ? ApplyBatchActionOptions
-    : ApplyActionOptions,
->(
-  args: A,
-  options?: OP,
-) => Promise<
-  ActionReturnTypeForOptions<OP>
->;
+/**
+ * Helper types for converting action definition parameter types to typescript types
+ */
+export namespace ActionParam {
```

Comment:
I know it may seem like I'm anti-namespace but I am actually okay with them if they are only types.